### ISSUE TITLE
char-cnn？

### ISSUE THREAD

**Issue by chenmozxh** [mkroutikov/tf-lstm-char-cnn#25]
`
conv = conv2d(input_, kernel_feature_size, 1, kernel_size, name="kernel_%d" % kernel_size)
def conv2d(input_, output_dim, k_h, k_w, name="conv2d"):
      with tf.variable_scope(name):
            w = tf.get_variable('w', [k_h, k_w, input_.get_shape()[-1], output_dim])
            b = tf.get_variable('b', [output_dim])
        return tf.nn.conv2d(input_, w, strides=[1, 1, 1, 1], padding='VALID') + b
`

should  k_w be dim of char-embedding, i.e. char_embed_size ?
1d-conv should be this, and your code is 2d-conv？

**Comment by mkroutikov** [comment]
First, a warning: This code is 2+ years old. Since then TF evolved from 0.8 to 2.0 !
TF also changed API and I am not sure this code runs on the modern TF.

I think I used 2-d convolution because the reference code was using it:

     https://github.com/carpedm20/lstm-char-cnn-tensorflow/blob/master/models/TDNN.py#L32

and Torch7 code is using SpacialConvolution (which, I believe is the cuddn analog of conv2d).

Ultimately, this code gave me the results that were matching the original article.

I am not saying the code has no bugs, just that it used to work reasonably well, and, as I remember, was compared to the original Torch7 code pretty closely and faithfully.

I also seem to vaguely remember that both 1d and 2d convolutions are used in text processing...
